Run doctests with tox

AndreMiras opened the pull request:
## What was wrong?

Currently the CI doesn't run doctests.
However keep in mind that now we have 5 tests failing.
If you're interested with the approach, I can also fix them in this PR or in another one.

Issue #9 

## How was it fixed?
Enabling doctests via `tox.ini` and `pytest.ini`

#### Cute Animal Picture

![image](https://user-images.githubusercontent.com/24973/47856647-fd339480-dde7-11e8-8b2f-1cd6e847934f.png)

pipermerriam commented:
Yes please, enabling doctest is something we've been adding to all of our repos and it should be in place here as well.

carver commented:
Yup, this is great, thanks!

> If you're interested with the approach, I can also fix them in this PR or in another one.

Please fix in this PR. As a rule we never merge a failing CI run.

FWIW, it's important that the readability of the docs not suffer in order to get the tests passing.

AndreMiras commented:
Great, then I may pick this up over the weekend if I get a chance.
Yes I agree the documentation should still be as readable :+1:

fubuloubu commented:
Seems like this can be closed?

carver commented:
> Seems like this can be closed?

I don't think this change was done yet. We are already test appropriately-marked snippets in the `docs/` folder. But this one would test the docstring'd method definitions, which is also reasonable to do.

AndreMiras commented:
Hi guys, I will eventually resume my contributions starting from this one as I have more time :tada: 
So I've rebased this PR, fixed the tests I could and skipped some others. Currently 9 are skipped, but I think it's a good start for now and we may reduce this number in the future.
I can't wait to contribute more in the future :smile:

AndreMiras commented on eth_account/account.py:
```diff
@@ -106,21 +106,21 @@ def decrypt(keyfile_json, password):
+            ...   'salt': '45cf943b4de2c05c2c440ef96af914a2'},
+            ...  'mac': 'f5e1af09df5ded25c96fcf075ada313fb6f79735a914adc8cb02e8ddee7813c3'},
+            ... 'id': 'b812f3f9-78cc-462a-9e89-74418aa27cb0',
+            ... 'version': 3}
```
Enabling doctests helped in detecting this "syntax error"

AndreMiras commented on eth_account/account.py:
```diff
@@ -106,21 +106,21 @@ def decrypt(keyfile_json, password):
 
             >>> import getpass
-            >>> Account.decrypt(encrypted, getpass.getpass())
+            >>> Account.decrypt(encrypted, getpass.getpass()) # doctest: +SKIP
```
I had to disable because obviously `getpass()` would ask for terminal input.
We could also give an example with an hardcoded value for the `password` that always produces an expected output.

carver replied:
Yeah, maybe worth doing both: demonstrating good practices but skipping the test (as you already did), and adding a version that can be doctest-ed.

AndreMiras replied:
Yes that could be it. Actually I initially wanted to find the password used to generate this output, but haven't found it so far.
I've created a task for adding a doctest that works https://github.com/ethereum/eth-account/issues/81

carver commented:
> Currently 9 are skipped

Maybe add an issue for fixing & re-enabling these tests?

carver commented on eth_account/messages.py:
```diff
@@ -187,13 +187,13 @@ def encode_defunct(
 
         # these four also produce the same hash:
-        >>> encode_defunct(w3.toBytes(text=message_text))
+        >>> encode_defunct(w3.toBytes(text=message_text)) # doctest: +SKIP
```
Just curious what the failure was here (I'm fine to push it to another issue)

AndreMiras replied:
```
NameError: name 'w3' is not defined
```

carver replied:
Hah, ok. Well that's not an instance method so it should be straightforward to switch `w3` to `Web3` and just add a `from web3 import Web3` up top. Just a note for when you/someone does the follow-up work.

AndreMiras commented:
Thanks for the review and approval @carver 

> I'll wait a bit for you to add that non-getpass() version. If you want to skip that for now, just let me know, and I'll merge it in.

I'm on holiday without much access to dev laptop, so I'd prefer we merge it. However I've created a [followup task](https://github.com/ethereum/eth-account/issues/81) that I'll be happy to pick up later.
